src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1186,6 +1186,18 @@ internal virtual void AddSynthesizedReturnTypeAttributes(PEModuleBuilder moduleB
 
         internal abstract bool IsNullableAnalysisEnabled();
 
+        /// <summary>
+        /// Gets the resolution priority of this method, 0 if not set.
+        /// </summary>
+        /// <remarks>
+        /// Do not call this method from early attribute binding, cycles will occur.
+        /// </remarks>
+        internal int OverloadResolutionPriority => CanHaveOverloadResolutionPriority ? (TryGetOverloadResolutionPriority() ?? 0) : 0;
+
+        internal abstract int? TryGetOverloadResolutionPriority();
+
+        internal abstract bool CanHaveOverloadResolutionPriority { get; }
```

Comment:
>CanHaveOverloadResolutionPriority [](http://example.com/codeflow?start=31&length=33)

Is `CanHaveOverloadResolutionPriority` only used to optimize callers in cases where we definitely know the method cannot use overload priority? If so, is the property necessary?

It looks like the property is only used in `MethodSymbol.OverloadResolutionPriority` and `OverloadResolution.RemoveLowerPriorityMembers()`. For `MethodSymbol.OverloadResolutionPriority`, could we check `MethodKind` instead? For `OverloadResolution.RemoveLowerPriorityMembers()`, we only call `CanHaveOverloadResolutionPriority` when there are at least two overloads, so there may not be much to optimize.

Reply:
I think you're right, so I'm greatly simplifying this.